src/librustc_mir/hair/cx/mod.rs

```diff
@@ -169,38 +156,46 @@ impl<'a, 'gcx, 'tcx> Cx<'a, 'gcx, 'tcx> {
         };
 
         let clamp = |n| {
-            let size = self.integer_bit_width(ty);
-            trace!("clamp {} with size {} and amt {}", n, size, 128 - size);
-            let amt = 128 - size;
+            let gcx = self.tcx.global_tcx();
+            let param_ty = self.param_env.and(gcx.lift(&ty).unwrap());
+            let bit_width = gcx.layout_of(param_ty).unwrap().size.bits();
```

Comment:
You can write `self.tcx` instead of `gcx` here. Which means you can use `self.tcx.lift_to_global(&ty)` above and not introduce a `gcx` variable.